sklearn/linear_model/tests/test_ridge.py

```diff
@@ -393,6 +393,40 @@ def _make_sparse_offset_regression(
     return X, y
 
 
+@pytest.mark.parametrize(
+    'solver', ['cholesky', 'sag', 'sparse_cg', 'lsqr', 'saga', 'ridgecv'])
+@pytest.mark.parametrize(
+    'n_samples,dtype,proportion_nonzero',
+    [(20, 'float32', .1), (40, 'float32', 1.), (20, 'float64', .2)])
+@pytest.mark.parametrize('sparse_X', [True, False])
+@pytest.mark.parametrize('seed', np.arange(3))
+def test_solver_consistency(
+        solver, proportion_nonzero, n_samples, dtype, sparse_X, seed):
+    accept_sparse = ['sparse_cg', 'ridgecv']
+    if sparse_X and solver not in accept_sparse:
+        pytest.skip()
```

Comment:
A better way would be something like
```py
@pytest.mark.parametrize(
    'solver, sparse_X',
    (solver, sparse_X in itertools.product(
       ['cholesky', 'sag', 'sparse_cg', 'lsqr', 'saga', 'ridgecv'], 
       [False, True]
     ) if not (sparse_X and solver not in accept_sparse)
)
```
otherwise this will yield aways skiped tests that will create uncessary noise in the global pytest output.